lib/metrics/StatsModel.js

```diff
@@ -143,6 +143,76 @@ class StatsModel extends StatsClient {
             return cb(null, statsRes);
         });
     }
+
+    /**
+     * normalize date timestamp to the nearest hour
+     * @param {Date} d - Date instance
+     * @return {number} timestamp - normalized to the nearest hour
+     */
+    _normalizeTimestampByHour(d) {
+        return d.setMinutes(0, 0, 0);
+    }
+
+    /**
+     * get previous hour to date given
+     * @param {Date} d - Date instance
+     * @return {number} timestamp - one hour prior to date passed
+     */
+    _getDatePreviousHour(d) {
+        return d.setHours(d.getHours() - 1);
+    }
+
+    /**
+     * get list of sorted set key timestamps
+     * @param {number} epoch - epoch time
+     * @return {array} array of sorted set key timestamps
+     */
+    getSortedSetHours(epoch) {
+        const timestamps = [];
+        let date = this._normalizeTimestampByHour(new Date(epoch));
+        while (timestamps.length < 24) {
+            timestamps.push(date);
+            date = this._getDatePreviousHour(new Date(date));
+        }
+        return timestamps;
+    }
+
+    /**
+     * get the normalized hour timestamp for given epoch time
+     * @param {number} epoch - epoch time
+     * @return {string} normalized hour timestamp for given time
+     */
+    getSortedSetCurrentHour(epoch) {
+        return this._normalizeTimestampByHour(new Date(epoch));
+    }
+
+    /**
+     * helper method to add element to a sorted set, applying TTL if new set
+     * @param {string} key - name of key
+     * @param {integer} score - score used to order set
+     * @param {string} value - value to store
+     * @param {callback} cb - callback
+     * @return {undefined}
+     */
+    addToSortedSet(key, score, value, cb) {
+        // milliseconds in a day
+        const msInADay = 24 * 60 * 60 * 1000;
```

Comment:
Minor: for readability, it seems like this should be within the if block in which it is used.